modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/controller/TaskController.scala

```diff
@@ -49,8 +50,13 @@ class TaskController {
   @GetMapping(value = Array("/{cid}/tasks"))
   @ResponseBody
   def getAll(@PathVariable("cid") cid: Int, req: HttpServletRequest, res: HttpServletResponse): List[Task] = {
-    authService.authorize(req, res)
-    taskService.getAll(cid)
+    val auth = authService.authorize(req, res)
+    val someCourseRole = courseRegistration.getParticipants(cid).find(_.user.id == auth.id).map(_.role)
```

Comment:
You should improve this code as it always has to search all users in the course. For this I would add a new method in the `courseRegistration` service which returns the role of a user in the course (for this you can orientate yourself on the function `getCoursePrivileges`).